internal/protocol/connector.go

```diff
@@ -119,13 +119,18 @@ func (c *Connector) Connect(ctx context.Context) (*Protocol, error) {
 // Make a single attempt to establish a connection to the leader server trying
 // all addresses available in the store.
 func (c *Connector) connectAttemptAll(ctx context.Context, log logging.Func) (*Protocol, error) {
+	// Wait here until we can lock the servers data
+	for {
+		if c.lock.TryLock() {
```

Comment:
I am concerned about this, it looks like a busy wait to me? What is this for? What was wrong with using c.lock.Lock()?